### DISCUSSION TITLE
Is LSP planned?

### DISCUSSION THREAD

**Discussion by glyh** [coalton-lang/coalton#990] [Ideas and Proposals]
One of the main benefit of using statically-typed system is that you get type error on the fly. So I think having an LSP will enhance user experience a lot.

**Comment by stylewarning** [comment]
Writing S-expressions in an S-expression aware editor becomes a must for most professional programmers. Emacs is the de facto standard in that department. As such, though this isn't certain, I expect we will add type hints and the like to SLIME before building an LSP server.

With that said, I don't think we've thought a lot about it beyond "Coalton DX has tremendous potential," and that's in more ways than just the editor experience.
